core/src/input_method.rs

```diff
@@ -80,36 +80,36 @@ pub enum Purpose {
 }
 
 impl InputMethod {
-    /// Merges two [`InputMethod`] strategies, prioritizing the first one when both open:
+    /// Merges two [`InputMethod`] strategies, prioritizing the first one when both allowed:
     /// ```
     /// # use iced_core::input_method::{InputMethod, Purpose, Preedit};
     /// # use iced_core::Point;
     ///
-    /// let open = InputMethod::Open {
+    /// let allowed = InputMethod::Allowed {
     ///     position: Point::ORIGIN,
     ///     purpose: Purpose::Normal,
     ///     preedit: Some(Preedit { content: "1".to_owned(), selection: None }),
     /// };
     ///
-    /// let open_2 = InputMethod::Open {
+    /// let allowed_2 = InputMethod::Allowed {
     ///     position: Point::ORIGIN,
     ///     purpose: Purpose::Secure,
     ///     preedit: Some(Preedit { content: "2".to_owned(), selection: None }),
     /// };
     ///
     /// let mut ime = InputMethod::Disabled;
     ///
-    /// ime.merge(&InputMethod::<String>::Allowed { position: Point::ORIGIN });
-    /// assert_eq!(ime, InputMethod::Allowed{ position: Point::ORIGIN });
+    /// ime.merge(&allowed);
+    /// assert_eq!(ime, allowed);
     ///
     /// ime.merge(&InputMethod::<String>::Disabled);
-    /// assert_eq!(ime, InputMethod::Allowed{ position: Point::ORIGIN });
+    /// assert_eq!(ime, allowed);
```

Comment:
Thank you fixing doctest I broke.

Merging allowed and open state made this doctest harder to understand, I feel.

Here original test showed how distinct states should be merged with the real use-case. Started with Disabled state and it becomes Allowed, then finally it changes to Open state.

But you lost that semantic here. This is the proof that this change make simpler the enum a bit, but it make less clear that mental model.

I found myself merging these two state will introduce another problem here.

https://github.com/iced-rs/iced/pull/2793#issuecomment-2644437375

Reply:
Okay, then let me focus on the first commit in this PR. I'll leave the fix for this part to you.